NIFI-6076 syslog5424 should support missing MSG

ottobackwards opened the pull request:
This issue is fixed in new simple-syslog-5424 v. 0.0.10
### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [-] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [-] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [-] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [-] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

bbende commented:
I don't mind merging this since it looks fine, but just out of curiosity, I ran the unit test that was added while still on the 0.0.8 dependency and it passed. Shouldn't that fail on 0.0.8 and pass on 0.0.10?

ottobackwards commented:
Please hold the merge and let me look at it.  In the jira I linked to the upstream issue, where I was able to reproduce the error.  Maybe I messed up moving the test over. If you are curious about the fix, I can provide a link to the upstream fix.

ottobackwards commented:
@bbende thanks for checking.  The issue I fixed is with the grammar.  It 'fails' in a different way depending on the structure of the message.  In the test I added, it _logically_ fails.  The second structured data ends up _being_ the message body, but the event is valid.
That is because there are two structured messages.
If there where one structured message, then it would fail parse with a missing space.

I need to add better tests.  Explicit tests.  I will do so and update.  Hope that makes sense, if you look at what I fixed in the grammar it should.

ottobackwards commented:
I'm going to close this pr while I work this out.

ottobackwards commented:
@bbende ok, I had more work to do upstream, thanks for the look.  Better tests, should be all set.

bbende commented:
@ottobackwards thanks! looks good, going to merge
